app/main.ipynb

```diff
@@ -560,42 +1917,71 @@
     "unknown_signature(lst, mask, new_value)\n",
     "lst"
    ],
-   "metadata": {
-    "collapsed": false,
-    "ExecuteTime": {
-     "end_time": "2023-07-03T14:11:19.696302Z",
-     "start_time": "2023-07-03T14:11:19.678876Z"
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "[0, 2, 0, 4, 0]"
+      ]
+     },
+     "execution_count": 44,
+     "metadata": {},
+     "output_type": "execute_result"
     }
-   }
+   ],
+   "execution_count": 44
   },
   {
    "cell_type": "code",
-   "execution_count": 30,
-   "outputs": [],
-   "source": [
-    "# write your numpy code here"
-   ],
    "metadata": {
     "collapsed": false,
+    "jupyter": {
+     "outputs_hidden": false
+    },
     "ExecuteTime": {
-     "end_time": "2023-07-03T14:09:42.667050Z",
-     "start_time": "2023-07-03T14:09:42.631627Z"
+     "end_time": "2024-12-27T18:39:01.196724Z",
+     "start_time": "2024-12-27T18:39:01.177508Z"
     }
-   }
+   },
+   "source": "np.array(lst, dtype=np.int8) * ~np.array(mask)",
```

Comment:
In Task 7.1, the NumPy implementation does not correctly apply the mask to update values in the array. You should use `np.where(mask, new_value, lst)` to replace elements in `lst` where `mask` is `True` with `new_value`.